lms/static/scripts/frontend_apps/index.tsx

```diff
@@ -90,8 +90,17 @@ export function init() {
 
   // Set route based on app mode.
   const routePath = routeForAppMode(config.mode);
+  const searchParams = new URLSearchParams(location.search);
   if (location.pathname !== routePath) {
     history.replaceState({}, 'unused', routePath);
+  } else if (config.mode === 'email-preferences' && searchParams.has('token')) {
+    // Remove token for email-preferences, if present in query string.
```

Comment:
I suggest to clarify that "token" refers to an auth token, which explains why we're stripping it. Also you can simplify URL construction using the `URL.searchParams` property.

This passes the tests in this PR:

```js
// Remove auth token for email-preferences if present. We originally tried
// to do this server-side but Safari failed to preserve cookies after the
// token-stripping redirect.
const url = new URL(location.href);
url.searchParams.delete('token');
history.replaceState({}, 'unused', url);
```